openrtb_ext/bid.go

```diff
@@ -7,8 +7,9 @@ import (
 
 // ExtBid defines the contract for bidresponse.seatbid.bid[i].ext
 type ExtBid struct {
-	Prebid *ExtBidPrebid   `json:"prebid,omitempty"`
-	Bidder openrtb.RawJSON `json:"bidder,omitempty"`
+	Prebid           *ExtBidPrebid   `json:"prebid,omitempty"`
+	Bidder           openrtb.RawJSON `json:"bidder,omitempty"`
+	AdServerCurrency string          `json:"ad_server_currency,omitempty"`
```

Comment:
I'm a little confused by this. First because I can't intuitively tell what it should be, and second because I can't think of a use case for it.

The OpenRTB spec has a spot for `cur` [on the BidResponse](https://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf#page=33). This looks like the same thing, but on individual `Bid` objects instead.

But... it looks like we're updating the `Bid.Price` (and other price-related fields) during the auction, before we return anything. Given that, is this supposed to return the _original_ Bid's currency, or the _final_ currency?

If it's the Bid's original currency, I'm not sure what the use case is. If it's the Bid's final currency, then it seems like we should just use `bidrequest.cur` instead.

Does that make sense? Or am I missing something?

Reply:
yeah, this is returning the original bid's currency from the bidder before any conversions are done. the final currency in the bid response's "cur" might be different depending on what the request had set, so this is just letting the requestor know the currency that the bidder returned.

Reply:
is there any use-case for it?

There's a fantastic talk on API design which I think every engineer should watch: https://www.youtube.com/watch?v=heh4OeB9A-c

The relevant point here is: APIs are very easy to add to, but very difficult (and sometimes impossible) to remove from. We should commit to as little behavior as possible until there're some compelling use-cases.

Reply:
totally agree. no use case for this as of now, so removing.